cli/src/main/java/com/devonfw/tools/ide/process/ProcessContextImpl.java

```diff
@@ -214,11 +214,19 @@ public ProcessResult run(ProcessMode processMode) {
    * @param is {@link InputStream}.
    * @return {@link CompletableFuture}.
    */
-  private static CompletableFuture<List<String>> readInputStream(InputStream is) {
+  private static CompletableFuture<List<String>> readInputStream(InputStream is, boolean errorStream, List<LogEvent> logs) {
 
     return CompletableFuture.supplyAsync(() -> {
 
       try (InputStreamReader isr = new InputStreamReader(is); BufferedReader br = new BufferedReader(isr)) {
+
+        String line;
+        while ((line = br.readLine()) != null) {
+          synchronized (logs) {
+            LogEvent logEvent = new LogEvent(errorStream, line);
+            logs.add(logEvent);
+          }
```

Comment:
I am not sure, if I am using `LogEvent` correctly. In this implementation, a single `LogEvent` instance represents one line from either `stout` or `sterr`. I don't see another solution to capture the entire log while keeping the order of the log streams.

Reply:
Now its `OutputMessage` instead of `LogEvent`. We can still discuss the name